iotserver/ServerThread.java

```diff
@@ -12,69 +14,31 @@ public ServerThread(Socket socket) {
         this.socket = socket;
     }
 
-    public void run() {
+    public void run() { 
         System.out.println("Accepted connection!");
 
         try {
             in = new ObjectInputStream(socket.getInputStream());
             out = new ObjectOutputStream(socket.getOutputStream());
+
+
         } catch (IOException e) {
             e.printStackTrace();
         }
     }
 
-    private void writeMessage(String message) {
-        out.writeInt(message.length());
-        byte buffer[] = message.getBytes();
-        out.write(buffer);
-    }
-
-    private String readMessage() {
-        int size = in.readInt();
-        byte buffer[] = in.readNBytes(size);
-        return new String(buffer);
+    private void writeMessage(MessageCode message) throws IOException {
```

Comment:
feels a little redundant, but we can get rid of these two as implementation goes on